### ISSUE TITLE
Windows.Forensics.BulkExtractor enhancement request

### ISSUE THREAD

**Issue by mpilking** [Velocidex/velociraptor#1829]
I would like to use the `FreeCommand` to limit the number of threads (`-j 1`) and also add a `find` regex string. The regex string has spaces in it, so it needs to be surrounded by double-quotes. However, when I add double-quotes in the `FreeCommand` field, backslashes are added to the command-line on Windows to escape it. Here's a resulting command captured from Sysmon:

```
C:\Users\SANSDFIR\AppData\Local\Temp\tmp460139415.exe -q 99999999999 -R -j 1 -E find -f \"type = mega\" -o C:\Users\SANSDFIR\AppData\Local\Temp\tmp441213208\F \\.\F:
```

I can't figure out how to remove those backslashes. If there's a trick to that, please let me know for future reference.

However, I think it might be good to also add another switch to add an option for the `FindRegex` variable to be added along with the `FreeCommand`. Something like the option to add `CarveEvtx` & `FindRegex` ([lines 57-60](https://github.com/Velocidex/velociraptor/blob/0ceb73cbc36deb814d397df8b9b4fd019f8a28f8/artifacts/definitions/Windows/Forensics/BulkExtractor.yaml#L132-L135)).

**Comment by mpilking** [comment]
The regex variable doesn't get passed into the command line though when the `FreeCommand` is used. 

So then you have to specify the regex in the `FreeCommand` field. If the regex has spaces, then you need the double-quotes, but Velociraptor is adding the backslash to escape the double-quotes, which invalidates the command.

**Comment by scudette** [comment]
Ah i see what you mean - the FreeCommand is actually a string which is unstructured and can be used to ad lib any command - it **is** actually passed to the shell and therefore it has escaping issues. 

You can see this tries to split the free form string into argv array using a naive split here
https://github.com/Velocidex/velociraptor/blob/0ceb73cbc36deb814d397df8b9b4fd019f8a28f8/artifacts/definitions/Windows/Forensics/BulkExtractor.yaml#L127

but since this artifact was written we now have a proper shell command line splitting function 
https://docs.velociraptor.app/vql_reference/plugin/commandline_split/

this function will take everything into account and produce a correct argv array including honor any potential shell escapes. It is probably a better alternative to just naive split()

**Comment by scudette** [comment]
Fixed in #1844
